File: OOP/PrototypeChain.py
class Vehicle:
    def move(self):
        return "Moving..."


class Car(Vehicle):
    def honk(self):
        return "Beep!"


car = Car()
print(isinstance(car, Car))
print(isinstance(car, Vehicle))
print(car.__class__ is Car)

car2 = Car()
print(car.honk.__func__ is car2.honk.__func__)

File: OOP/PrototypeChain.js
class Vehicle {
  move() { return "Moving..."; }
}

class Car extends Vehicle {
  honk() { return "Beep!"; }
}

const car = new Car();

console.log(car instanceof Car);        // true
console.log(car instanceof Vehicle);    // true
console.log(Object.getPrototypeOf(car) === Car.prototype); // true

// All cars share the same method reference
const car2 = new Car();
console.log(car.honk === car2.honk);   // true — same prototype method